react-auth-client/src/sdk/api.js

```diff
@@ -13,26 +13,19 @@ import config from "../config";
- * @param clientId a string that specifies the application’s UUID.
- * @param redirectUri a string that specifies the URL that specifies the return entry point of the application. This is a required property.
- * @param scope a string that specifies permissions that determine the resources that the application can access. This parameter is not required, but it is needed to specify accessible resources.
  * @param state a string that specifies an optional parameter that is used to maintain state between the logout request and the callback to the endpoint specified by the post_logout_redirect_uri query parameter.
  * @param nonce a string that is used to associate a client session with an ID token, and to mitigate replay attacks. The value is passed through unmodified from the authentication request to the ID token.
- * @param prompt a string that specifies whether the user is prompted to login for re-authentication. The prompt parameter can be used as a way to check for existing authentication, verifying that the user is still present for the current session. For prompt=none, the user is never prompted to login to re-authenticate, which can result in an error if authentication is required. For prompt=login, if time since last login is greater than the max-age, then the current session is stashed away in the flow state and treated in the flow as if there was no previous existing session. When the flow completes, if the flow’s user is the same as the user from the stashed away session, the stashed away session is updated with the new flow data and persisted (preserving the existing session ID). If the flow’s user is not the same as the user from the stashed away session, the stashed away session is deleted (logout) and the new session is persisted.
- * @param maxAge a string that specifies the maximum amount of time allowed since the user last authenticated. If the max_age value is exceeded, the user must re-authenticate.
- * @param acr_values an optional parameter that designates whether the authentication request includes steps for a single-factor or multi-factor authentication flow. This parameter maps to the name of a sign-on policy that must be assigned to the application. For more information, see Sign-on policies.
- * @returns {Promise<T | never>}
  */
-const authorize = (environmentId, responseType='token id_token', clientId, redirectUri, scope,
-    state, nonce, prompt = 'login', maxAge, acrValues = null) => {
+const authorize = (state, nonce) => {
   let authUrl = `${getBaseApiUrl(
-      true)}/${environmentId}/as/authorize?` +
-      `response_type=${responseType}&client_id=${clientId}&redirect_uri=${redirectUri}&prompt=${prompt}` +
-      (scope ? `&scope=${scope}` : '') +
-      (maxAge ? `&max_age=${maxAge}` : '') +
-      (acrValues ? `&acr_values=${acrValues}` : '') +
+      true)}/${config.environmentId}/as/authorize?` +
+      `response_type=${config.responseType ? config.responseType
```

Comment:
nit: the code would read easier if you made a const for the response type and set it using that

Reply:
Since I have a couple of ternary operators, and to not extract const for all of them I refactored to 
```
let authUrl = `${getBaseApiUrl(true)}/${config.environmentId}/as/authorize?` +
      `client_id=${config.clientId}&` +
      `redirect_uri=${config.redirectUri}&`+
      `response_type=${config.responseType ? config.responseType: 'token id_token'}` +
      (config.prompt ? `&prompt=${config.prompt}` : '') +
      (config.scope ? `&scope=${config.scope}` : '') +
      (config.maxAge ? `&max_age=${config.maxAge}` : '') +
      (config.acrValues ? `&acr_values=${config.acrValues}` : '') +
```
What do you say?

Reply:
I can read that better, thanks!